### ISSUE TITLE
RawUI & PrivateData:  Console Colors are broken?

### ISSUE THREAD

**Issue by JontuKontar** [microsoft/terminal#372]
When using `(Get-Host).PrivateData` or `(Get-Host).UI.RawUI` to set Console Colors, they only exist when the console is opened.

Typed text is not bound to them nor do the properties of `PrivateData` or `RawUI` indicate that they are properly set.  (`Clear-Host`, effectively removes all changes from the Console.)

Enabling "Use Legacy Console" restores expected function.

Not sure if there is something else I should be doing to make the color changes stick with the "Modern Console" or if this is a legitimate bug in the way console colors are applied to the Console.

**Comment by oising** [comment]
This issue belongs on github.com/powershell/powershell

**Comment by zadjii-msft** [comment]
If this is a regression between the v1 (legacy) and v2 consoles, we're happy to take a look, but we'll need a minimal repro that we can use to debug this.

________________________________

**Comment by lawlz** [comment]
Hey All!

I am having the same issue and I believe it is related to the psreadline v2, which you probably already know, since there are many issues around console coloring, [#818](https://github.com/lzybkr/PSReadLine/issues/818), [#774](https://github.com/lzybkr/PSReadLine/issues/774), #276   .  Here is a reproduction of what I am seeing:

![image](https://user-images.githubusercontent.com/6010549/55277322-d01edb80-52cc-11e9-88ae-9749beca776b.png)

See how the foregroundcolor reverts to an initial or default background color, and potentially because they are writing `CSI 0 m` and not saving or persisting this new console color setting.  Comment found here from the [main maintainer](https://github.com/Microsoft/console/issues/276#issuecomment-432106661).

After downloading the latest module from the above repo, it seems to be working, but the versioning is weird.  It appears the psreadline module in the repo is still 1.0 and the version that is shipped with windows is installing the 2.0, which appears to be beta...  Also, any reference to the 2.0 Beta still seems to tie back to the repo above which has the 1.0 versioning...  Here is a [reference to 2.0](https://github.com/lzybkr/PSReadLine/issues/854#issuecomment-477851836), but the only activity from SteveL, externally facing that is, shows a forked repo of the above PSReadline.

Here it is working from the package downloaded from that repo:

![errorRepro](https://user-images.githubusercontent.com/6010549/55278464-a66cb100-52da-11e9-95fb-39c55107d4ef.PNG)

Long story short, I believe this to be a duplicate issue and fixed in a 'new' version of PSReadline, unless the goal is to move away from the 1.0 and go to 2.0.  If that is the case, the latest build for 2.0, that I could find, was last updated [last September and still a beta](https://www.powershellgallery.com/packages/PSReadLine/2.0.0-beta3) and still has this issue.  I am not sure, but wanted to update this issue and hopefully get some insider insight on what the direction forward is on this module and maybe why MS would ship a beta version of a module when, on this machine, I don't have developer mode on or anything.  Thank you.

**Comment by DHowett-MSFT** [comment]
> maybe why MS would ship a beta version of a module

If I had to guess, it's because PSReadline 2.0 (even in beta) is so much better than 1.4 that it was worth it in their risk assessment.

Since it looks like this issue was originally caused by PSReadline saving and restoring colors, I'm going to mark this one for garbage collection. If it _isn't_ because PSReadline was saving and restoring colors, please feel free to reopen it.

(A quick check: `Remove-Module PSReadline`, then do your color test. If they still don't stay, it could be our fault.)

**Comment by lawlz** [comment]
Thank you so much for the reply and information.  Yes it works fine once you `remove-module psreadline`, but of course you lose all that PSReadLine goodness.  

As an aside, I pulled down the 2.0.0-Beta4 version that was released to the [Gallery 19 days ago](https://www.powershellgallery.com/packages/PSReadLine/2.0.0-beta4) and it appears to have the required patch to fix this issue!  

Thanks again for your help Dustin!
